One test is failing, you will need to run `./test/partest --update-check test/files/run/reflection-fieldmirror-ctorparam.scala` after confirming the new wording is as intended.

```
BUILD FAILED
/home/jenkins/workspace/scala-2.12.x-validate-test/build.xml:1483: The following error occurred while executing this line:
/home/jenkins/workspace/scala-2.12.x-validate-test/build-ant-macros.xml:776: Test suite finished with 1 case failing:
fail - run/reflection-fieldmirror-ctorparam.scala  [output differs]% scalac reflection-fieldmirror-ctorparam.scala
% /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java \
  -Dfile.encoding=UTF-8 \
  -server \
  -XX:+AggressiveOpts \
  -XX:+UseParNewGC \
  -Xmx2G \
  -Xss1M \
  -XX:MaxPermSize=512M \
  -XX:ReservedCodeCacheSize=128M \
  -Dpartest.threads=4 \
  -classpath \
  /home/jenkins/workspace/scala-2.12.x-validate-test/test/files/run/reflection-fieldmirror-ctorparam-run.obj:/home/jenkins/workspace/scala-2.12.x-validate-test/build/pack/lib/scala-library.jar:/home/jenkins/workspace/scala-2.12.x-validate-test/build/pack/lib/scala-reflect.jar:/home/jenkins/workspace/scala-2.12.x-validate-test/build/pack/lib/scala-compiler.jar:/home/jenkins/workspace/scala-2.12.x-validate-test/build/pack/lib/scalap.jar:/home/jenkins/workspace/scala-2.12.x-validate-test/m2repo/org/scala-lang/modules/scala-partest_2.12.0-M2/1.0.9/scala-partest_2.12.0-M2-1.0.9.jar:/home/jenkins/workspace/scala-2.12.x-validate-test/m2repo/com/googlecode/java-diff-utils/diffutils/1.3.0/diffutils-1.3.0.jar:/home/jenkins/workspace/scala-2.12.x-validate-test/m2repo/org/scala-sbt/test-interface/1.0/test-interface-1.0.jar:/home/jenkins/workspace/scala-2.12.x-validate-test/m2repo/org/scala-lang/modules/scala-xml_2.12.0-M2/1.0.5/scala-xml_2.12.0-M2-1.0.5.jar:/home/jenkins/workspace/scala-2.12.x-validate-test/m2repo/org/scala-lang/modules/scala-parser-combinators_2.12.0-M2/1.0.4/scala-parser-combinators_2.12.0-M2-1.0.4.jar:/home/jenkins/workspace/scala-2.12.x-validate-test/m2repo/org/scalacheck/scalacheck_2.12.0-M2/1.11.6/scalacheck_2.12.0-M2-1.11.6.jar:/home/jenkins/workspace/scala-2.12.x-validate-test/build/pack/lib/scala-partest-extras.jar:/home/jenkins/workspace/scala-2.12.x-validate-test/build/pack/lib/scala-partest-javaagent.jar:/home/jenkins/workspace/scala-2.12.x-validate-test/test/files/lib/annotations.jar:/home/jenkins/workspace/scala-2.12.x-validate-test/test/files/lib/enums.jar:/home/jenkins/workspace/scala-2.12.x-validate-test/test/files/lib/genericNest.jar:/home/jenkins/workspace/scala-2.12.x-validate-test/test/files/lib/jsoup-1.3.1.jar:/home/jenkins/workspace/scala-2.12.x-validate-test/test/files/lib/macro210.jar:/home/jenkins/workspace/scala-2.12.x-validate-test/test/files/lib/methvsfield.jar:/home/jenkins/workspace/scala-2.12.x-validate-test/test/files/lib/nest.jar \
  -Dfile.encoding=UTF-8 \
  -Djava.library.path=/home/jenkins/workspace/scala-2.12.x-validate-test/test/files/run \
  -Dpartest.output=/home/jenkins/workspace/scala-2.12.x-validate-test/test/files/run/reflection-fieldmirror-ctorparam-run.obj \
  -Dpartest.lib=/home/jenkins/workspace/scala-2.12.x-validate-test/build/pack/lib/scala-library.jar \
  -Dpartest.reflect=/home/jenkins/workspace/scala-2.12.x-validate-test/build/pack/lib/scala-reflect.jar \
  -Dpartest.comp=/home/jenkins/workspace/scala-2.12.x-validate-test/build/pack/lib/scala-compiler.jar \
  -Dpartest.cwd=/home/jenkins/workspace/scala-2.12.x-validate-test/test/files/run \
  -Dpartest.test-path=/home/jenkins/workspace/scala-2.12.x-validate-test/test/files/run/reflection-fieldmirror-ctorparam.scala \
  -Dpartest.testname=reflection-fieldmirror-ctorparam \
  -Djavacmd=/usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java \
  -Djavaccmd=javac \
  -Duser.language=en \
  -Duser.country=US \
  scala.tools.nsc.MainGenericRunner \
  -usejavacp \
  Test \
  jvm > reflection-fieldmirror-ctorparam-run.log
% diff /home/jenkins/workspace/scala-2.12.x-validate-test/test/files/run/reflection-fieldmirror-ctorparam-run.log /home/jenkins/workspace/scala-2.12.x-validate-test/test/files/run/reflection-fieldmirror-ctorparam.check
@@ -1,3 +1,3 @@
-class scala.ScalaReflectionException: Scala field x of type class A isn't represented as a Java field, neither does it
-have a Java accessor method. Note that private parameters of class constructors don't get mapped
-onto fields and/or accessors, unless they are used outside of their declaring constructors.
+class scala.ScalaReflectionException: Scala field x isn't represented as a Java field, neither it has a Java accessor method
+note that private parameters of class constructors don't get mapped onto fields and/or accessors,
+unless they are used outside of their declaring constructors.
```

--- a/src/reflect/scala/reflect/runtime/JavaMirrors.scala
+++ b/src/reflect/scala/reflect/runtime/JavaMirrors.scala
@@ -124,9 +124,9 @@ private[scala] trait JavaMirrors extends internal.SymbolTable with api.JavaUnive
     private def ErrorArrayConstructor(sym: Symbol, owner: Symbol) = abort(s"Cannot instantiate arrays with mirrors. Consider using `scala.reflect.ClassTag(<class of element>).newArray(<length>)` instead")
     private def ErrorFree(member: Symbol, freeType: Symbol)       = abort(s"cannot reflect ${member.kindString} ${member.name}, because it's a member of a weak type ${freeType.name}")
     private def ErrorNonExistentField(sym: Symbol)                = abort(
-      sm"""Scala field ${sym.name} isn't represented as a Java field, neither it has a Java accessor method
-          |note that private parameters of class constructors don't get mapped onto fields and/or accessors,
-          |unless they are used outside of their declaring constructors.""")
+      sm"""Scala field ${sym.name} of ${sym.owner} isn't represented as a Java field, nor does it have a
+          |Java accessor method. One common reason for this is that it may be a private class parameter
+          |not used outside the primary constructor.""")
 
     /** Helper functions for extracting typed values from a (Class[_], Any)
      *  representing an annotation argument.

--- a/test/files/run/reflection-fieldmirror-ctorparam.check
+++ b/test/files/run/reflection-fieldmirror-ctorparam.check
@@ -1,3 +1,3 @@
-class scala.ScalaReflectionException: Scala field x isn't represented as a Java field, neither it has a Java accessor method
-note that private parameters of class constructors don't get mapped onto fields and/or accessors,
-unless they are used outside of their declaring constructors.
+class scala.ScalaReflectionException: Scala field x of class A isn't represented as a Java field, nor does it have a
+Java accessor method. One common reason for this is that it may be a private class parameter
+not used outside the primary constructor.

